src/librustc_session/session.rs

```diff
@@ -55,6 +57,52 @@ pub enum CtfeBacktrace {
     Immediate,
 }
 
+/// New-type wrapper around `usize` for representing limits. Ensures that comparisons against
+/// limits are consistent throughout the compiler.
+#[derive(Clone, Copy, Debug)]
+pub struct Limit(usize);
```

Comment:
Is there a reason not to just make the `usize` a public field?

Reply:
Fixed!